app/src/main/java/com/futurice/freesound/feature/search/SearchViewModel.java

```diff
@@ -88,23 +90,23 @@ void openSoundDetails(@NonNull final Sound sound) {
     }
 
     @Override
-    public void bind(@NonNull final CompositeSubscription subscriptions) {
+    public void bind(@NonNull final CompositeDisposable disposables) {
         searchTermRelay.subscribeOn(Schedulers.computation())
                        .observeOn(Schedulers.computation())
                        .map(String::trim)
-                       .switchMap(this::searchOrClear)
+                       .debounce(2, TimeUnit.SECONDS)
+                       .switchMap(it -> searchOrClear(it).toObservable())
                        .observeOn(mainThread())
                        .subscribe(nothing1(),
                                   e -> Timber.e(e, "Error when setting search term"));
     }
 
-    private Observable<Unit> searchOrClear(@NonNull final String s) {
+    private Maybe<Unit> searchOrClear(@NonNull final String s) {
         return TextUtils.isNullOrEmpty(s) ?
-                searchDataModel.clear() :
-                Observable.just(s)
-                          .filter(StringFunctions.isNotEmpty())
-                          .debounce(2, TimeUnit.SECONDS)
-                          .switchMap(searchDataModel::querySearch);
+                searchDataModel.clear().toMaybe() :
+                Single.just(s)
+                      .filter(StringFunctions.isNotEmpty())
+                      .flatMap(it -> searchDataModel.querySearch(it).toMaybe());
```

Comment:
This used to be a `switchMap` - any reason to change to a `flatmap`? Actually, given that it's a `Single` it shouldn't make a difference in terms of output, only performance if `switchMap` is faster than `flatmap` for example.